src/buildercore/bootstrap.py

```diff
@@ -478,6 +478,13 @@ def update_ec2_stack(stackname, context, concurrency=None, formula_revisions=Non
         master_builder_key = download_master_builder_key(stackname)
 
     def _update_ec2_node():
+        # upload private key if not present remotely
+        if not files.exists("/root/.ssh/id_rsa", use_sudo=True):
+            # if it also doesn't exist on the filesystem, die horribly.
+            # regular updates shouldn't have to deal with this.
+            pem = stack_pem(stackname, die_if_doesnt_exist=True)
```

Comment:
Two things perplex me:
1. we download the right key at https://github.com/elifesciences/builder/pull/360/files#diff-bfa1d9cba2806d6d2d41031884d3e00eR478 so it should really use that rather than `stack_pem`?
2. It should only do this if `is_masterless` is `True`, otherwise we randomly upload some key to all stacks

Reply:
And we are already doing both 1 and 2 in https://github.com/elifesciences/builder/pull/360/files#diff-bfa1d9cba2806d6d2d41031884d3e00eR504